test/controllers/user_notifications_controller_test.rb

```diff
@@ -6,6 +6,8 @@ class UserNotificationsControllerTest < ActionController::TestCase
   setup :login_and_be_admin, :create_items
 
   test 'routes' do
+    assert_equal user_user_notifications_url(@user),
```

Comment:
Why is this added?

Reply:
cuz i wan't to prove that user_user_notifications_url(@user) is the right link instead of user_user_notification_path(@user). I can remove it if you want

Reply:
I understand your intentions, and I think it's good that you have an eye for testing. Nevertheless here are several reasons why I think this is not a good idea -
1. You included `test.host` here which might change in different environments. If this host name somehow changes (for example, maybe upgrading Rails somehow changes it), this test will break.
2. We have tested `user_user_notifications_path` below, which is essentially the same.
3. The exact URL, which you are testing here, is actually not relevant to the logic of the unsubscribe function. We don't necessarily care which exact link the unsubscribe link is pointing to - we only care that it points to the one under `user_user_notifications_url`.

Given these considerations I suggest you remove this test.

Reply:
I got it, thanks for the enlightenment